mempool/transaction.go

```diff
@@ -0,0 +1,11 @@
+package mempool
+
+// Tx represents a read-only interface for the transactions.
+type Tx interface {
```

Comment:
We should define a custom pkg for tx `tx.Transaction`. Alternatively, we can abstract transaction to messages, but it might be confusing,